### ISSUE TITLE
UnusedImport false positive when extension method import is only used in documentation

### ISSUE THREAD

**Issue by matejdro** [detekt/detekt#6540]
## Expected Behavior

UnusedImports should not flag extension method import that is only used in documentation

## Observed Behavior

UnusedImport falsly flags it

## Steps to Reproduce

Analyze following code with `UnusedImports` enabled:

```kotlin
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.onStart

/**
 * [Flow.onStart]
 */
class Test
```

## Your Environment

* Version of detekt used: 1.23.1
* Version of Gradle used (if applicable): 8.4

**Comment by cortinico** [comment]
`UnusedImports` is coming from the `detekt-formatting` ruleset which is the KtLint wrapper. So it means that this is a bug in the KtLint code. Have you searched there and/or opened an issue over there?

**Comment by matejdro** [comment]
Ah right, sorry, I mixed up the two rules. Made the ktlint issue here: https://github.com/pinterest/ktlint/issues/2330

Do you want me to close this issue?

**Comment by BraisGabin** [comment]
Yes, we can close it.

**Comment by matejdro** [comment]
Wait, are we sure that UnusedImports is from ktlint?

https://github.com/detekt/detekt/issues/6560#issuecomment-1775900550 I am getting confused now :sweat_smile:

**Comment by cortinico** [comment]
`UnusedImport` is from the `style` ruleset (== from detekt):
https://detekt.dev/docs/next/rules/style/#unusedimport

`NoUnusedImports` is from the `formatti` ruleset (== from ktlint):
https://detekt.dev/docs/next/rules/formatting/#nounusedimports

**Comment by matejdro** [comment]
Can we reopen this ticket then? Since it's not from ktlint

**Comment by matejdro** [comment]
It's not waiting an author's feedback. I might be able to create PR for that at some point, but I cannot give any time lines.

**Comment by atulgpt** [comment]
Hi @matejdro I have created one PR https://github.com/detekt/detekt/pull/6567 this is pending review

**Comment by matejdro** [comment]
Thank you!

**Comment by matejdro** [comment]
@cortinico @BraisGabin It appears this is back in the `2.0.0-alpha.1`. Can we reopen this? Or do you want me to open a new issue?

**Comment by cortinico** [comment]
> Can we reopen this? Or do you want me to open a new issue?

Let's open a new issue 👍

**Comment by matejdro** [comment]
Sorry, my bad, the issue was not related to detekt. It looks like I had a duplicate comment somehow:

https://github.com/inovait/kotlinova/commit/6dc7fc81eaa9b6ee5ecfc6e6b0e559164d6b46e2

Interestingly enough, old Detekt did not catch this, but 2.0 does.
